Initial draft of the Partial Fanout changes in the spec

vrom911 opened the pull request:
Resolves https://github.com/cardano-scaling/hydra/issues/2304
Fixes https://github.com/cardano-scaling/hydra/issues/2274

vrom911 committed:
Initial draft of the Partial Fanout changes in the spec

vrom911 committed:
Write clearer about bls

vrom911 committed:
Add limits and BLS papers

vrom911 committed:
Highlight new changes in RED

vrom911 committed:
Add accumulator to snapshot

As close to https://github.com/cardano-scaling/hydra/pull/2324 as possible

vrom911 committed:
Add my name into authors

ch1bo commented on src/Hydra/Protocol/Preliminaries.tex:
```diff
@@ -37,6 +37,7 @@ \subsection{Notation}
 	\item $\concat : \tyBytes^* \to \tyBytes$ is concatenating bytes, we also use operator $\bigoplus$ for this
 	\item $\hash : x \to \tyBytes$ denotes a collision-resistant
 	      hashing function and $x^{\#}$ indicates the hash of $x$
+	      \textcolor{red}{\todo{Update hash function usage to support partial fanout where needed}}
```
Isn't this what this PR should be about?

ch1bo reviewed (changes requested):
Good starting point!

I like that you added a section to the preliminaries, but it is not fully consistent yet. My biggest feedback is to separate `fanout` and `partialFanout` as I'm missing validator checks about the leftover accumulator in the outgoing `Closed` datum.

vrom911 committed:
Separate fanout and partial fanout

vrom911 committed:
Unify Closed state accumulators using BLS accCombine operation

Simplify the protocol by combining η, η_αΔ, η_ωΔ into a single
unified accumulator using BLS composition during close transition.
This reduces complexity in the Closed state and fanout procedures.
